Includes/config.cpp

```diff
@@ -47,37 +49,60 @@ void from_json(nlohmann::json const& j, ftpConfig& f) {
 }
 
 void ftpConfig::setEnabled(bool val) {
-  if (val != enable) {
+  if (enable != val) {
     enable = val;
-    changed = true;
   }
 }
 
 void ftpConfig::setUser(std::string const& user) {
-  if (!user.compare(user)) {
+  if (!username.compare(user)) {
     username = user;
-    changed = true;
   }
 }
 
 void ftpConfig::setPassword(std::string const& pwd) {
-  if (!this->password.compare(pwd)) {
-    this->password = pwd;
-    changed = true;
+  if (!password.compare(pwd)) {
+    password = pwd;
   }
 }
 
 void ftpConfig::setPort(int p) {
-  if (this->port != p) {
-    this->port = p;
-    changed = true;
+  if (port != p) {
+    port = p;
   }
 }
 
 Config::Config() {
   std::ifstream configFile(HOME "config.json");
   nlohmann::json json;
   configFile >> json;
-  FTP = json["settings"]["ftp"].get<ftpConfig>();
+  if (json.contains("settings")) {
+    settings = json["settings"].get<Settings>();
+  } else {
+    changed = true;
+  }
+  if (!json.contains("menu")) {
+    json["menu"] = R"([{"label": "Games", "type": "scan", "path": "F:\\Games\\"},
+        {"label": "Applications", "type": "scan", "path": "F:\\Apps"},
+        {"label": "Launch DVD", "type": "launch", "path": "D:\\default.xbe"},
+        {"label": "Settings", "type": "settings"},
+        {"label": "Reboot", "type": "reboot"}])"_json;
+    changed = true;
+  }
   menu = json["menu"];
+  configFile.close();
+}
+
+void Config::setChanged() {
+  changed = true;
+}
+
+void Config::storeToDisk() {
+  if (changed) {
+    std::ofstream configFile(HOME "config.json");
```

Comment:
Is it safe to assume that the folder where NevolutionX is installed is writable?
Consider using `E:\UDATA\deadbeef\` or similar instead.

Reply:
Someone _could_ run it from a DVD :s in which case it would certainly not be writable.

Reply:
True, true. From what I can tell, this implementation doesn't break anything on write failure - might as well keep it like this (at least for now).